### PR TITLE
boards: nxp: mimxrtxxx: fix non-optimal sector distribution

### PR DESCRIPTION
**Pull request by butok**
- Optimize slot sizes for MCUBoot swap move algorithm for mimxrt595/685_evk boards.
- Use DT_SIZE_K/M macros for slot sizes.

### CODE REVIEW

**Review by decsny — APPROVED** [review]

**Review by danieldegrasse — APPROVED** [review]

**Review by EmilioCBen — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

File: boards/nxp/mimxrt595_evk/mimxrt595_evk_mimxrt595s_cm33.dts [hunk 1]
```diff
@@ -407,17 +407,20 @@ zephyr_udc0: &usbhs {
...
+			slot1_partition: partition@322000 {
 				label = "image-1";
-				reg = <0x00321000 DT_SIZE_K(3072)>;
+				reg = <0x00382000 DT_SIZE_M(3)>;
 			};
-			storage_partition: partition@621000 {
+			storage_partition: partition@622000 {
 				label = "storage";
-				reg = <0x00621000 DT_SIZE_M(57)>;
+				reg = <0x00682000 (DT_SIZE_M(58) - DT_SIZE_K(520))>;
```
Comment (dleach02) [inline boards/nxp/mimxrt595_evk/mimxrt595_evk_mimxrt595s_cm33.dts:423]: Why this change? Trying to understand the -520K from the base size?

Comment (butok) [reply boards/nxp/mimxrt595_evk/mimxrt595_evk_mimxrt595s_cm33.dts:423]: It is based on: 64M /*Flash size*/ - 6M /*slot0&1*/ - (128K+98*4K) /* MCUBoot, swap status& move*/
